### DISCUSSION TITLE
How does incremental static generation work?

### DISCUSSION THREAD

**Discussion by mem88** [vercel/next.js#39790] [Help]
If I have a page with `[id].js`

where I only pre build top 1k ids (out of 10 million ids)  in`getStaticPaths`, what happens when next.js sees a new id

does `getStaticProps` still get called or will I get 404?

do I have to set any parameter to make server rending of new ids possible?

**Comment by icyJoseph** [comment]
Hi,

You need to tell `getStaticPaths` what to do by using the `fallback` property. Refer to this part of the documentation, https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-pages, where `fallback` is set to `true`, meaning that at runtime, when a `new id` is seen, Next.js will generate the page, while showing a fallback is necessary.

Note that `fallback: true` requires you to account for a view of your page, where the result of `getStaticProps` passed to the page is `undefined`. 

There's also `fallback: "blocking"` to just show a blank and skip the above, but it also gives poorer user experience. By default Next.js skips the fallback page for bots, and uses blocking.
